handbook/direction/code-graph/index.md

```diff
@@ -0,0 +1,33 @@
+# Code Graph
+
+Code Graph is responsible for search (the user experience as well as performance), insights, code intelligence, and batch changes. This page describes overarching initiatives that tie multiple teams and milestones within Code Graph together.
+
+Individual team direction pages can be found on our [product home page](../../product/index.md#code-graph). For details on how the themes here become reality, as well as what each team is working on next for their individual visions, take a look there.
+
+## Mission
+
+What is the team's reason for existing? What is the fundamental value they aim to provide? Who are your key customers/stakeholders and why, and are there any customers/stakeholders that you're explicitly choosing not to address that people might have expected you to?
+
+## Vision
+
+### 1 Year vision
+
+> Add a sentence or two here indicating what the most important things you want to achieve/problems you want to solve in the next year are. These should in most cases move your three and ten year visions forward.
+
+### 3 Year vision
+
+> Add a sentence or two here describing what your product will be like in three years, also focusing on the problems you'll solve and the value you'll deliver. This should be grounded in reality, but three years is a lot of time - be ambitious!
+
+### 10 year vision
+
+> A sentence or two on your ten year vision is where you can really go wild and describe the problems you want to solve and solutions you want to deliver unbound by limitations of technology or scope. Be inspiring and look to the our [10 and 30 year strategies](../../company/strategy.md#sourcegraph-strategy) for ideas.
+
+## Themes
```

Comment:
Product maturity would be a good theme as well. 

On search specifically:
* Commit and diff search remain un-indexed, thus:
  * performance suffers
  * code monitors are limited to 50 repositories, where companies want to monitor their entire code base.
* Structural search does not function at scale
* Contexts are not yet discoverable
* Notebooks show great promise but require 

We are addressing these issues now, but it will be an ongoing effort for some time.

Reply:
Great one, added @rrhyne